src/Umbraco.Core/Media/UploadAutoFillProperties.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using Umbraco.Core.Configuration;
+using Umbraco.Core.Configuration.UmbracoSettings;
+using Umbraco.Core.IO;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Models;
+
+namespace Umbraco.Core.Media
+{
+    /// <summary>
+    /// Provides methods to manage auto-fill properties for upload fields.
+    /// </summary>
+    internal static class UploadAutoFillProperties
```

Comment:
Refactoring so that most of these classes are not static anymore. Will still use singletons here and there because it's a pain to remove them and so easy in v8.